pkg/operator/sync.go

```diff
@@ -461,10 +461,16 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig, _ *configv1.ClusterOpera
 		}
 	}
 
-	// sync up CAs
-	rootCA, err := optr.getCAsFromConfigMap("kube-system", "root-ca", "ca.crt")
+	// Sync up machine-config-server-ca bundle
+	// Attempt the managed bundle in the MCO namespace first. If not found, use the unmanaged rootCA bundle instead
+	machineConfigServerCABundle, err := optr.getCAsFromConfigMap(ctrlcommon.MCONamespace, ctrlcommon.MachineConfigServerCAName, "ca-bundle.crt")
 	if err != nil {
-		return err
+		if apierrors.IsNotFound(err) {
+			machineConfigServerCABundle, err = optr.getCAsFromConfigMap("kube-system", ctrlcommon.RootCAConfigMapName, "ca.crt")
```

Comment:
I think we briefly talked about this already, but I'm wondering what happens if you keep both CAs around merged together?

Reply:
I think the `machine-config-server-ca` configmap is a better indicator of currently valid CAs. This check is really just meant as a safeguard to use the legacy `root-ca` configmap until our cert controller kicks in for the first time. Once the `machine-config-server-ca` configmap exists, the MCS TLS cert will also have been updated, and whatever is in `root-ca` configmap won't be valid anymore. I expect this to happen immediately when the MCC comes up on installation, or during an upgrade to an OCP version where the MCO does cert management.